libs/vscode/tasks/src/lib/nx-task-commands.ts

```diff
@@ -341,15 +354,58 @@ async function promptForList() {
 
 async function promptForMigrate() {
   const telemetry = getTelemetry();
-  telemetry.featureUsed('affected-cli');
-  const task = NxTask.create(
-    {
-      command: 'migrate',
-      flags: [],
-    },
-    cliTaskProvider.getWorkspacePath()
+  telemetry.featureUsed('migrate command');
+
+  const { workspacePath } = await getNxWorkspace();
+
+  const packageJson = await readAndParseJson(
+    join(workspacePath, 'package.json')
   );
-  tasks.executeTask(task);
+
+  if (
+    packageJson.dependencies == undefined ||
+    packageJson.devDependencies == undefined
+  ) {
+    return;
+  }
+
+  const dependencyToMigrate = await new Promise<string | undefined>(
+    (resolve) => {
+      const quickPick = window.createQuickPick();
+      quickPick.title = 'Select dependency';
+      quickPick.items = buildQuickPickItems(packageJson);
+      quickPick.placeholder = 'Select the dependency you want to migrate.';
+
+      quickPick.onDidAccept(() => {
+        resolve(quickPick.selectedItems[0]?.label);
+        quickPick.hide();
+      });
+
+      quickPick.show();
+    }
+  );
+
+  if (dependencyToMigrate === undefined) {
+    return;
+  }
+
+  const depVersioningInfo = await resolveDependencyVersioning(
+    dependencyToMigrate
+  );
+
+  if (depVersioningInfo !== undefined) {
```

Comment:
i mean this is suuuuuuuuper nit picky but if you were to keep with the above coding style, it would be more consistent like this :P 
```suggestion
  if (depVersioningInfo !== undefined) {
  	return
 }
```

Reply:
Used it only because other functions in the same file use similar approach :)